qemu/tests/vsock_test.py

```diff
@@ -0,0 +1,152 @@
+import os
+import time
+import logging
+import aexpect
+
+from avocado.utils import process
+
+from virttest import data_dir
+from virttest import error_context
+from virttest import utils_misc
+
+
+def compile_nc_vsock(vm, session, test):
+    """
+    Copy and compile nc-vsock on both host and guest
+
+    :param vm: Object qemu_vm.VM
+    :param session: vm session
+    :param test: QEMU test object
+    :return: Path to binary nc-vsock or None if compile failed
+    """
+    nc_vsock_dir = '/home/'
+    nc_vsock_bin = 'nc-vsock'
+    nc_vsock_c = 'nc-vsock.c'
+    src_file = os.path.join(data_dir.get_deps_dir("nc_vsock"), nc_vsock_c)
+    bin_path = os.path.join(nc_vsock_dir, nc_vsock_bin)
+    rm_cmd = 'rm -rf %s*' % bin_path
+    session.cmd(rm_cmd)
+    process.system(rm_cmd, shell=True, ignore_status=True)
+    cmd_cp = "cp %s %s" % (src_file, nc_vsock_dir)
+    process.system(cmd_cp)
+    vm.copy_files_to(src_file, nc_vsock_dir)
+    compile_cmd = "cd %s && gcc -o %s %s" % (
```

Comment:
@vivianQizhu how to ensure gcc is installed?  Does guest or host  kickstart configuration file include this package?

Reply:
@xiangchunfu Can not ensure that, when no gcc installed, the case will raise error, I think that's enough. @yanan-fu Any comment about that? Thanks.

Reply:
@vivianQizhu @xiangchunfu   It is ok, when test with internal kar, we can ensure it with both host and guest.
For host, bootstrap will install it,  and for guest, kickstart have this operation too.